src/documentation/ptransform-style-guide.md

```diff
@@ -0,0 +1,488 @@
+---
+layout: default
+title: "PTransform Style Guide"
+permalink: /documentation/ptransform-style-guide/
+---
+
+# PTransform style guide
+
+## A style guide for writers of new reusable PTransform's
+
+{:toc}
+
+# Language-neutral considerations
+
+## Consistency
+Be consistent with prior art:
+
+* Adhere to [Beam design principles](https://beam.apache.org/contribute/design-principles/).
```

Comment:
Please make all links relative; use `baseurl` and other jekyll syntax. This will prevent staging to work or republishing the page at any other location.